### PR TITLE
RELATED: SD-559 add filter context type

### PR DESCRIPTION
**Pull request by naibinh**

### CODE REVIEW

**Comment by yenkins** [comment]
Currently destined for [release R188](https://confluence.intgdc.com/display/Dev/R188).
This release is **zero downtime** release.
Code drop: 2019-08-09 00:00.

**Comment by yenkins** [comment]
Story [SD-559](http://jira.intgdc.com/browse/SD-559) gooddata-js - Enhance to adapt new PDF export endpoint
	Status: New
	Priority: Major
	Assignee: Binh Nguyen

**Comment by yenkins** [comment]
## client-libs/gooddata-typings-pull-request-compliance-check check failed
### require-evaluated-risk
Ticket should have regression risk evaluated before merging, see [GoodConfluence](https://confluence.intgdc.com/display/office/Labels+In+JIRA#LabelsInJIRA-QA&DELI)
- SD-559

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-check/28/ : UNSTABLE in 11s
- gooddata-typings-unit-tests-zuul-docker-el7 : SKIPPED

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-check/29/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/99/ : SUCCESS in 1m 13s

**Comment by yenkins** [comment]
Story [SD-559](http://jira.intgdc.com/browse/SD-559) gooddata-js - Enhance to adapt new PDF export endpoint
	Status: Hacking
	Priority: Major
	Assignee: Binh Nguyen

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-check/30/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/100/ : SUCCESS in 55s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-check/31/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/101/ : SUCCESS in 53s

File: src/FilterContext.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// (C) 2019 GoodData Corporation
+
+export type FilterContextItem = IAttributeFilter | IDateFilter;
```
Comment (lanhq) [inline src/FilterContext.ts]: These Types look like conflict with https://github.com/gooddata/gooddata-typings/blob/master/src/AFM.ts#L118

Comment (naibinh) [reply src/FilterContext.ts]: I don't think so.
This type must be refelcted to backend [specification](https://github.com/gooddata/gdc-webapp/blob/develop/specification/md/obj.res#L1752)

File: src/FilterContext.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        attributeElements: string[];
+    };
+}
+
+export interface IDateFilter {
+    dateFilter: {
+        type: 'relative' | 'absolute';
+        granularity: string;
+        dataSet?: string;
+        from?: string;
```
Comment (lanhq) [inline src/FilterContext.ts]: In both relative and absolute type, `from`, `to` is needed. Which case can `from` and `to` be missing?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-check/32/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/102/ : SUCCESS in 1m 13s

**Comment by lanhq** [comment]
LGTM

File: src/FilterContext.ts (hunk 2 above)
Comment (naibinh) [reply src/FilterContext.ts]: For `relative`, this year or today means `from=to=0`, so we can omit it.
For `absolute`, it means from/to point to `current` time.
Anyway, FE should send this info explicitly to BE.
Fixed.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-check/33/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/103/ : SUCCESS in 33s

**Comment by yenkins** [comment]
Currently destined for [release R189](https://confluence.intgdc.com/display/Dev/R189).
This release is **zero downtime** release.
Code drop: 2019-08-23 00:00.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-check/34/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/104/ : FAILURE in 44s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-check/35/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/105/ : SUCCESS in 58s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-check/36/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/107/ : SUCCESS in 20s

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@gooddata/typings",
-  "version": "2.16.0",
+  "version": "2.16.0-alpha-naibinh-nab-sd-559-filterContext-type-2019-08-09T07-47-16-281Z",
```
Comment (ptquang86) [inline package.json]: Do we really need this?

Comment (naibinh) [reply package.json]: No, we don't. My bad actually.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-check/37/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/108/ : SUCCESS in 32s

**Comment by ptquang86** [comment]
LGTM

**Comment by yenkins** [comment]
## client-libs/gooddata-typings-pull-request-compliance-gate check failed
### require-approval
- pull request must be approved

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-gate/13/ : UNSTABLE in 11s
- gooddata-typings-unit-tests-zuul-docker-el7 : SKIPPED

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-gate/14/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/109/ : SUCCESS in 1m 18s

**Pull request merged**